rubicon/objc/objc.py

```diff
@@ -1342,14 +1347,69 @@ def __new__(cls, object_ptr):
 
         return objc_instance
 
+    def __getitem__(self, key):
+        classname = objc.class_getName(self.__dict__['objc_class'].ptr).decode('utf-8')
+        if classname.startswith('__NSArray') or classname.startswith('__NSSingleObjectArray'):
+            if isinstance(key, int):
+                # Support negative indices as indexing from the right
+                if key < 0:
+                    key = self.count - (-1 * key)
+                if key > (self.count - 1) or key < 0:
+                    raise IndexError('list index out of range')
+                else:
+                    return self.objectAtIndex_(key)
+            elif isinstance(key, slice):
+                array = ObjCClass('NSMutableArray').array()
+                helper = [item for item in self]
+                for item in helper[key]:
+                    array.addObject_(item)
```

Comment:
This design decision needs discussion:

If `isinstance(a, NSArray)`, would you rather `a[:]` return a Python list or clone `NSArray`?